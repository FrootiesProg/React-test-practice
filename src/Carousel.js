import React, { useState } from "react";
import Card from "./Card";
import "./Carousel.css";

function Carousel({ photos, title }) {
  const [currCardIdx, setCurrCardIdx] = useState(0);
  const isOnFirstImage = currCardIdx === 0;

  if (!photos || photos.length === 0) {
    return (
      <div className="Carousel">
        <h1>{title}</h1>
        <p>No photos to display.</p>
      </div>
    );
  }

  const total = photos.length;
  const isOnLastImage = currCardIdx === total - 1;

  // Move to the previous image (decrement currCardIdx)
  function goBack() {
    if (currCardIdx > 0) {
      setCurrCardIdx(currCardIdx - 1);
    }
  }

  // Move to the next image (increment currCardIdx)
  function goForward() {
    if (currCardIdx < total - 1) {
      setCurrCardIdx(currCardIdx + 1);
    }
  }

  return (
    <div className="Carousel">
      <h1>{title}</h1>
      <div className="Carousel-main">
        <i
          className={`bi bi-arrow-left-circle ${
            isOnFirstImage ? "disabled" : ""
          }`}
          onClick={goBack}
          data-testid="left-arrow"
        />
        <Card
          caption={photos[currCardIdx].caption}
          src={photos[currCardIdx].src}
          currNum={currCardIdx + 1}
          totalNum={total}
        />
        <i
          className={`bi bi-arrow-right-circle ${
            isOnLastImage ? "disabled" : ""
          }`}
          onClick={goForward}
          data-testid="right-arrow"
        />
      </div>
    </div>
  );
}

export default Carousel;
