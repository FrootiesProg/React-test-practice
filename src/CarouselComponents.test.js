import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Carousel from "./Carousel";

const TEST_IMAGES = [
  { id: 1, src: "image1.jpg", caption: "Caption 1" },
  { id: 2, src: "image2.jpg", caption: "Caption 2" },
];

test("Left arrow is missing on first image", () => {
  const { getByTestId } = render(
    <Carousel photos={TEST_IMAGES} title="Image Carousel" />
  );

  const leftArrow = getByTestId("left-arrow");
  const rightArrow = getByTestId("right-arrow");

  // Check that the left arrow is missing on the first image
  expect(leftArrow).toBeDisabled();

  // Click the right arrow to move to the second image
  fireEvent.click(rightArrow);

  // Check that the left arrow is now enabled on the second image
  expect(leftArrow).not.toBeDisabled();
});

test("Right arrow is missing on last image", () => {
  const { getByTestId } = render(
    <Carousel photos={TEST_IMAGES} title="Image Carousel" />
  );

  const leftArrow = getByTestId("left-arrow");
  const rightArrow = getByTestId("right-arrow");

  // Click the right arrow to move to the second image
  fireEvent.click(rightArrow);

  // Check that the right arrow is enabled on the second image
  expect(rightArrow).not.toBeDisabled();

  // Click the right arrow again to move to the last image
  fireEvent.click(rightArrow);

  // Check that the right arrow is missing on the last image
  expect(rightArrow).toBeDisabled();
});
