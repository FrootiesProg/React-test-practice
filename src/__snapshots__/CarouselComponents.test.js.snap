// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel component snapshot 1`] = `
<div
  className="Carousel"
>
  <h1>
    Image Carousel
  </h1>
  <div
    className="Carousel-main"
  >
    <i
      className="bi bi-arrow-left-circle"
      onClick={[Function]}
    />
    <div
      className="Card"
    >
      <h4
        className="Card-title"
      >
        Caption 1
      </h4>
      <img
        alt="Caption 1"
        className="Card-image"
        src="image1.jpg"
      />
      <small
        className="Card-small"
      >
        Image 
        1
         of 
        2
        .
      </small>
    </div>
    <i
      className="bi bi-arrow-right-circle"
      onClick={[Function]}
    />
  </div>
</div>
`;
